Vizzuality/marxan-cloud
# users and projects API modules - scaffolding

## Scaffolding for API Users and Projects modules

### Overview

Basic functionality for users and projects modules. Incomplete, mostly to be able to test JSON:API serialization.

### Designs

N/A

### Testing instructions

```
make clean-slate # CAREFUL! This will delete the Docker volume for the PostgreSQL for API container, after stopping all the services
make start
```

All the migrations should run, from scratch.

Once all the services are up again:

```
make seed-api-with-test-data
```
This will import test data (roles, users and projects, as well as associating users on projects through specific roles).

Then:

* `GET /users` should return a list of users (in JSON:API format - but with `Content-Type: application/json`)
* `GET /projects` should return a list of projects, referencing users associated to them via roles

### Feature relevant tickets

https://www.pivotaltracker.com/story/show/176206971
https://www.pivotaltracker.com/story/show/176376639

---

## Checklist before submitting

      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
